modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -1122,6 +1448,102 @@ private User _assertLockout(
 		return user;
 	}
 
+	private void _assertUserCreatedWithPasswordPolicy(
+			boolean ldapUser, String password)
+		throws Exception {
+
+		try {
+			User user = _createUser(ldapUser, password);
+
+			Assert.assertEquals(
+				"User was created with incorrect LDAP Server Id",
+				ldapUser ? 1 : -1, user.getLdapServerId());
+			Assert.assertTrue(
+				"During creation, user did not have password policy applied",
+				user.isPasswordReset());
+			Assert.assertNotNull(
+				"User is bypassing portal password policy",
+				user.getPasswordPolicy());
+		}
+		catch (Throwable throwable) {
+			throw new Exception("Unable to create user", throwable);
+		}
+	}
+
+	private void _assertUserPasswordException(boolean ldapUser, String password)
+		throws Exception {
+
+		try {
+			_createUser(ldapUser, password);
+
+			Assert.fail("Password policy is not being applied to user");
+		}
+		catch (Throwable throwable) {
+			if (!(throwable instanceof UserPasswordException)) {
+				throw new Exception(
+					"Exception unrelated to password policy was thrown",
+					throwable);
+			}
+		}
```

Comment:
Can you catch the specific exception you're expecting and let others bubble up?

Reply:
Also, as before, avoid mixing logic (user creation) with assertions in util methods. Why not use `AssertUtils.assertFailure` in the tests that invoke this and remove this method?
```java
AssertUtils.assertFailure(
        UserPasswordException.class,
        "Password policy is not being applied to user",
        () -> _createUser(ldapUser, password);
```